### PR TITLE
Integrate Maven builds and GH actions setup

### PR DESCRIPTION
**Pull request by xSAVIKx**
In this PR I have copied over all the relevant Maven and GH actions-related changes I introduced in my fork:
* Added [Maven wrapper ](https://maven.apache.org/wrapper/) to simplify local maven setup.
* Changed Maven repos, license, links, etc. to reflect changes.
* Swapped Travis builds with GH Actions. Publishing requires extra secure variables we can discuss and configure separately.

### CODE REVIEW

**Comment by xSAVIKx** [comment]
@jlouvel, can you please review this? Please let me know how can I share the JCA.

**Comment by jlouvel** [comment]
Thanks a lot @xSAVIKx this looks great!
I'll let @thboileau do a more thorough review and merge the PR as soon as possible.
For the JCA, could you send it to jlouvel@gmail.com?

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Set default behavior to automatically normalize line endings.
```
Comment (thboileau) [inline .gitattributes:1]: :+1:

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+* text=auto
+
+# Force batch scripts to always use CRLF line endings so that if a repo is accessed
+# in Windows via a file share from Linux, the scripts will work.
+*.{cmd,[cC][mM][dD]} text eol=crlf
+*.{bat,[bB][aA][tT]} text eol=crlf
+*.{ics,[iI][cC][sS]} text eol=crlf
```
Comment (thboileau) [inline .gitattributes]: hum, I wonder why we should take such files into consideration?
I'm not an expert, but `ics` files are calendar files, aren't they?

File: .gitattributes [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+# Force bash scripts to always use LF line endings so that if a repo is accessed
+# in Unix via a file share from Windows, the scripts will work.
+*.sh text eol=lf
```
Comment (thboileau) [inline .gitattributes:11]: :+1:

File: pom.xml [hunk 4]
```diff
@@ -10,38 +10,36 @@
...
     <inceptionYear>2005</inceptionYear>
 
     <scm>
-        <url>http://restlet.tigris.org/svn/restlet/modules/</url>
+        <url>https://github.com/restlet/restlet-framework-java.git</url>
+        <connection>scm:git:${project.scm.url}</connection>
+        <developerConnection>scm:git:${project.scm.url}</developerConnection>
     </scm>
 
     <repositories>
```
Comment (thboileau) [inline pom.xml]: I think we can drop the `repositories` section. The`Restlet repository` is useful only for 2.4 and below versions of the the framework.

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    branches:
+      - "2.5"
+  pull_request:
+  workflow_dispatch:
+concurrency:
+  group:
+    ${{ github.repository }}-${{ github.workflow }}-${{ github.event.number || github.head_ref || github.run_id || github.sha }}
+  cancel-in-progress: true
+env:
+  JAVA_VERSION: 11
```
Comment (thboileau) [inline .github/workflows/build.yaml:14]: do you mind to step back to version `8`? we changed our mind

File: .github/workflows/publish.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+on:
+  release:
+    types: [ created ]
+  workflow_dispatch:
+concurrency:
+  group:
+    ${{ github.repository }}-${{ github.workflow }}-${{ github.event.number || github.head_ref || github.run_id || github.sha }}
+  cancel-in-progress: true
+env:
+  JAVA_VERSION: 11
```
Comment (thboileau) [inline .github/workflows/publish.yaml:11]: do you mind to step back to version `8`? we changed our mind

Comment (thboileau) [inline .github/workflows/publish.yaml]: do you mind to step back to version `8`? we changed our mind

```suggestion
  JAVA_VERSION: 8
```

File: .github/workflows/build.yaml (hunk 5 above)
Comment (thboileau) [inline .github/workflows/build.yaml]: do you mind to step back to version `8`? we changed our mind

```suggestion
  JAVA_VERSION: 8
```

File: .gitignore [hunk 7]
```diff
@@ -1,12 +1,122 @@
...
-/modules/org.restlet.test/velocity.log.1
-.DS_Store
 *.iml
 .DS_Store
 .idea
 target
 .classpath
 .project
 .settings
+*.class
```
Comment (thboileau) [inline .gitignore:8]: :+1:

File: README.md [hunk 8]
```diff
@@ -1,5 +1,13 @@
 # Restlet Framework
 
+
```
Comment (thboileau) [inline README.md:3]: :+1:

File: pom.xml [hunk 9]
```diff
@@ -10,38 +10,36 @@
...
             <id>maven-restlet</id>
             <name>Restlet repository</name>
-            <url>https://maven.restlet.org</url>
+            <url>https://maven.restlet.talend.com</url>
         </repository>
     </repositories>
 
     <properties>
+        <release>false</release>
+        <java.version>11</java.version>
```
Comment (thboileau) [inline pom.xml]: do you mind to step back to version `8`? we changed our mind

```suggestion
        <java.version>1.8</java.version>
```

File: pom.xml [hunk 10]
```diff
@@ -95,54 +93,229 @@
...
+                <property>
+                    <name>publish</name>
+                    <value>github</value>
+                </property>
+            </activation>
+            <distributionManagement>
+                <repository>
+                    <id>github</id>
+                    <name>GitHub Packages</name>
+                    <url>https://maven.pkg.github.com/xSAVIKx/restlet-framework-java</url>
```
Comment (thboileau) [inline pom.xml]: Should be `restlet`?

```suggestion
                    <url>https://maven.pkg.github.com/restlet/restlet-framework-java</url>
```

**Review by thboileau — COMMENTED** [review]
Hi @xSAVIKx 
many, many thanks for your great contribution.
I have a few comments. Could you have a look?

File: .gitattributes (hunk 2 above)
Comment (xSAVIKx) [reply .gitattributes]: This is just an auto-generated template. I can drop that particular one if you prefer.

File: pom.xml (hunk 4 above)
Comment (xSAVIKx) [reply pom.xml]: I believe it should be stated here while you're still gonna publish there as well, right?

**Comment by xSAVIKx** [comment]
@thboileau please take a look again. I believe I've answered/resolved all the open points

File: .gitattributes (hunk 2 above)
Comment (thboileau) [reply .gitattributes]: Hi @xSAVIKx 
do you mind to drop it?

File: pom.xml (hunk 4 above)
Comment (thboileau) [reply pom.xml]: oh, I have to tell you we want to publish to Maven Central for 2.5 versions and higher. That's why I'm inclined to drop the repositories section.
Does it sound correct to you?

**Comment by thboileau** [comment]
hi @xSAVIKx 
I've added some comments

File: .gitattributes (hunk 2 above)
Comment (xSAVIKx) [reply .gitattributes]: removed 👍

File: pom.xml (hunk 4 above)
Comment (xSAVIKx) [reply pom.xml]: removed 👍

**Comment by xSAVIKx** [comment]
@thboileau resolved 👍

**Review by thboileau — APPROVED** [review]
Many thanks again @xSAVIKx !!

**Comment by xSAVIKx** [comment]
Thx for the review. FYI, I don't have a way to merge this, so that's on you guys.

**Pull request merged**